server/protocols/webadmin-cli/src/main/java/org/apache/james/httpclient/FeignClientFactory.java

```diff
@@ -0,0 +1,43 @@
+/******************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one     *
+ * or more contributor license agreements.  See the NOTICE file   *
+ * distributed with this work for additional information          *
+ * regarding copyright ownership.  The ASF licenses this file     *
+ * to you under the Apache License, Version 2.0 (the              *
+ * "License"); you may not use this file except in compliance     *
+ * with the License.  You may obtain a copy of the License at     *
+ *                                                                *
+ * http://www.apache.org/licenses/LICENSE-2.0                     *
+ *                                                                *
+ * Unless required by applicable law or agreed to in writing,     *
+ * software distributed under the License is distributed on an    *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY         *
+ * KIND, either express or implied.  See the License for the      *
+ * specific language governing permissions and limitations        *
+ * under the License.                                             *
+ ******************************************************************/
+
+package org.apache.james.httpclient;
+
+import feign.Feign;
+
+public class FeignClientFactory {
+    // Some JWT related fields here!
+    private String jwt;
+
+    public FeignClientFactory(String jwt) {
+        this.jwt = jwt;
+    }
+
+    public Feign.Builder builder() {
+        if (this.jwt == null) {
```

Comment:
Code smell. Never ever never use null. 

Idea 1: Use Optional to explicitly handle the fact that it is missing.

Idea 2: Use an interface for FeignClientFactory, and have two implementation (one with jwt, one without) so that we don't need to check for nullity every time we create a feign client.